integration/kubernetes/confidential/agent_image.bats

```diff
@@ -152,4 +162,7 @@ setup() {
 
 teardown() {
 	teardown_common
+	remove_test_image "$image_unsigned_protected" || true
```

Comment:
So we should remove all images in `teardown()` like this: 
```
 remove_test_image "$image_cosigned" || true
 remove_test_image "$image_cosigned_other" || true
 remove_test_image "$image_simple_signed" || true
 remove_test_image "$image_signed_protected_other" || true
 remove_test_image "$image_unsigned_protected" || true
 remove_test_image "$image_unsigned_unprotected" || true
 remove_test_image "$image_authenticated" || true

```

Reply:
Even with the CI job **only pulling the image on the guest**?

Reply:
An image is associated with a snapshotter. Therefore, prior to utilizing the snapshotter, it's necessary to remove all tested images. Otherwise, the image pull request will not be directed to nydus-snapshotter, as containerd interprets the image as already available. So we should remove all tested images before running the tests.

Reply:
Otherwise, an error occurs: 
```
Warning  FailedCreatePodSandBox  3s    kubelet   Failed to create pod sandbox: 
rpc error: code = NotFound desc = failed to create containerd container: error unpacking image: 
failed to extract layer sha256:1021ef88c7974bfff89c5a0ec4fd3160daac6c48a075f74cff721f85dd104e68: 
failed to get reader from content store: content digest sha256:fbe1a72f5dcd08ba4ca3ce3468c742786c1f6578c1f6bb401be1c4620d6ff705: not found

```

Reply:
> If we neglect to remove the images, an error will occur when creating the container:

This is a huge drawback, IMHO.

It forces users to cleanup all the images on their systems before using nydus, making this approach only okay for a clean installation. :-/